onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -1201,24 +1174,22 @@ class PoolOpBuilder : public BaseOpBuilder {
   static void CreateSharedOpBuilder(const std::string& op_type, OpBuilderRegistrations& op_registrations);
 
  private:
+  static bool IsQuantizedOp(const NodeUnit& node_unit);  // TODO, see if we want to move this to BaseOpBuilder
   Status AddToModelBuilderImpl(ModelBuilder& model_builder, const NodeUnit& node_unit) const override ORT_MUST_USE_RESULT;
 };
 
+/* static */ bool PoolOpBuilder::IsQuantizedOp(const NodeUnit& node_unit) {
+  // TODO, add support for QDQ NodeUnit
+  return node_unit.OpType() == "QLinearAveragePool";
```

Comment:
As this is an internal op I don't think you'd ever see it in a valid onnx model in QDQ format. It's only added if you optimized to extended level.

Reply:
Same for other things like QLinearSigmoid. We should never see any contrib ops in a QDQ format model.

Reply:
Yes, for now we only consider the single_node NodeUnit, will need to add extra logic to handle QDQ NodeUnit such as DQ->Sigmoid->Q